src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -953,6 +953,72 @@ void SQLStorage::clearInstalledVersions() {
   }
 }
 
+void SQLStorage::storeInstallationResult(const data::OperationResult& result) {
+  SQLite3Guard db(config_.sqldb_path.c_str());
+
+  if (db.get_rc() != SQLITE_OK) {
+    LOG_ERROR << "Can't open database: " << db.errmsg();
+    return;
+  }
+
+  auto statement = db.prepareStatement<std::string>(
```

Comment:
annoying, the extra arguments compared to the template parameters is not caught at compile time. Only at run time, and not with every sqlite version!

Reply:
Foolish mistake, but yes, very strange that it compiled and all the tests passed on my machine!

Reply:
Yes, there is something with the template argument deduction I suspect. Probably not an issue with sqlite version, like I suspected, but something like different gcc interpretations depending on version...